sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -115,30 +146,6 @@ private boolean shouldApplyScopeData(
       }
     }
 
-    if (event != null) {
-      if (event.getThrowable() != null
-          && options.containsIgnoredExceptionForType(event.getThrowable())) {
-        options
-            .getLogger()
-            .log(
-                SentryLevel.DEBUG,
-                "Event was dropped as the exception %s is ignored",
-                event.getThrowable().getClass());
-        options
-            .getClientReportRecorder()
-            .recordLostEvent(DiscardReason.EVENT_PROCESSOR, DataCategory.Error);
-        return SentryId.EMPTY_ID;
-      }
-      event = executeBeforeSend(event, hint);
-
-      if (event == null) {
-        options.getLogger().log(SentryLevel.DEBUG, "Event was dropped by beforeSend");
-        options
-            .getClientReportRecorder()
-            .recordLostEvent(DiscardReason.BEFORE_SEND, DataCategory.Error);
-      }
-    }
-
     SentryId sentryId = SentryId.EMPTY_ID;
```

Comment:
We could early return here before building and sending the envelope if the event has been dropped. This way we wouldn't send any attachments but also wouldn't send the session update immediately. It may be better to just send the session update and skip the attachment(s) if the event was dropped.

Reply:
Created this PR https://github.com/getsentry/sentry-java/pull/2002 to show how this might work.

Reply:
Right now this is a bit confusing, I'd not mix up this PR with session update inconsistencies and attachments.
I'm actually wondering why attachments are being mentioned in this PR, mind clarifying?

Reply:
Got it, I missed the point that https://github.com/getsentry/sentry-java/issues/1916 mentions the attachments as well, and I agree that does not make sense.